src/account/twoFactor/useTwoFaToggleMutation.js

```diff
@@ -0,0 +1,81 @@
+import { useDataMutation, useAlert } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { useMemo, useState } from 'react'
+import userProfileStore from '../../profile/profile.store.js'
+
+const mutationBase = {
+    type: 'create',
+    data: ({ code }) => ({ code }),
+}
+const enableMutation = {
+    ...mutationBase,
+    resource: '/2fa/enabled',
+}
+const disableMutation = {
+    ...mutationBase,
+    resource: '/2fa/disabled',
+}
+
+const getAlertMessage = ({ nextIsTwoFaEnabled, error }) => {
+    if (nextIsTwoFaEnabled) {
+        if (error) {
+            return (
+                error?.message ??
+                i18n.t('Could not enable 2 Factor Authentication')
+            )
+        } else {
+            return i18n.t('Two factor authentication was enabled successfully')
+        }
+    } else {
+        if (error) {
+            return (
+                error?.message ??
+                i18n.t('Could not disable two factor authentication')
+            )
+        } else {
+            return i18n.t('Two factor authentication was disabled successfully')
+        }
+    }
+}
+
+const getAlertOptions = ({ error }) =>
+    error ? { critical: true } : { success: true }
+
+export default function useTwoFaToggleMutation({ isTwoFaEnabled }) {
+    const [
+        lastActionWasTwoFaDisableSuccess,
+        setLastActionWasTwoFaDisableSuccess,
+    ] = useState(false)
+    const { show: showAlert } = useAlert(getAlertMessage, getAlertOptions)
```

Comment:
I don't think this is a big issue (and maybe it's just a UX choice, in which case I would be quite 🙊), but it seems that the use of this `showAlert` function for all of these different alerts leads to them being replaced by one another rather than stacked on top of one another. So, like if I successfully enable and then disable my 2FA in succession, I only end up with the one success message about disabling...and the text just changes in the existing alert, so it's hard to see as a new alert.

Realistically, I don't think people would be making changes in such rapid succession that it really matters, but I just point it out (if this wasn't already the intent)